### ISSUE TITLE
s3p rm --bucket --prefix --filter

### ISSUE THREAD

**Issue by kirubakarrs** [generalui/s3p#57]
It'd be nice to have a remove command to delete files in parallel. I have been trying to delete large number of files matching the `Key` or `LastModified` which takes a very long time using AWS CLI.

**Comment by siebrand** [comment]
I delete files not in target with up to 1000 objects in a single command by processing the s3p output. Here is the bash scripting that I haven't cleaned up. Hope this helps you.

```
#!/usr/bin/env bash
time npx s3p sync --storage-class INTELLIGENT_TIERING --bucket $SOURCEBUCKET --to-bucket $TARGETBUCKET --overwrite --max-queue-size 20000 --list-concurrency 50 >> $LOGFILE 2>&1
echo "DEBUG: Format all to be deleted objects as \"Key\": \"Object\""
TO_DELETEFILE="/var/log/to_delete.txt"
grep 'aws s3 rm' $LOGFILE | cut -d' ' -f4 |sed "s,s3://$TARGETBUCKET/,," |awk '{print "{\"Key\": \"" $0 "\"},"}' > $TO_DELETEFILE
DATE_TODAY=$(date "+%Y-%m-%d")
echo "DEBUG: Create directory"
mkdir -p to_delete_$DATE_TODAY
echo "DEBUG: Move into directory"
cd to_delete_$DATE_TODAY
echo "DEBUG: Create files with 1000 lines with paths of to be deleted objects. 1000 is the limit for a single delete command"
split -l 1000 $TO_DELETEFILE to_delete_
echo "DEBUG: Remove the last comma from each file to make it valid JSON"
echo "DEBUG: Prepend: {\"Objects\":["
echo "DEBUG: Append: ,\"Quiet\":true]}"
for i in `find -type f` ; do sed -i -e '$ s/.$/],"Quiet":true}/' -e '1s/^/{"Objects":[/' $i; done
echo "DEBUG: Loop all delete batches"
for i in `find -type f` ; do echo "DEBUG: Process $i"; aws s3api delete-objects --bucket $TARGETBUCKET --delete file://$i; done
```

**Comment by kirubakarrs** [comment]
This helps, thanks!
